### PR TITLE
Adjust snippet for configuration cache

### LINKED ISSUE #21803: CCH22-6: Adjust the `snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule` sample for configuration cache
The sample project located under [`subprojects/docs/src/snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule`](https://github.com/gradle/gradle/tree/master/subprojects/docs/src/snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule) does not meet [the configuration cache requirements](https://docs.gradle.org/nightly/userguide/configuration_cache.html#config_cache:requirements).

The build scripts in the `groovy` and `kotlin` parts of the sample project need to be adjusted to follow the requirements.
### Context
This issue is a part of the [Gradle Configuration Cache Hackathon 2022](https://github.com/gradle/cc-hackathon-2022).

If you wish to participate and contribute the fix, please follow [the suggested workflow](https://github.com/gradle/cc-hackathon-2022/blob/main/workflow.md).
### Expected behavior

- The sample project is fixed so that the related documentation tests pass with configuration cache enabled.
- The tests related to this sample project are removed from the `testsToBeFixedForConfigurationCache` list in `subprojects/docs/build.gradle`. You can find the test IDs below.
- The doc pages that include the snippets from this sample project might need to be updated to be consistent with the sample. Feel free to ask the team for help with this! You can find the pages listed below.
### Test failure details
These tests running in `:docs:docsTest` check the sample project and are failing with configuration cache enabled:
* `snippet-dependency-management-customizing-resolution-capability-substitution-rule_groovy_capability`[(Build Scan)](https://ge.gradle.org/s/wkszl3ueqiln2/tests/:docs:docsTest/org.gradle.docs.samples.Bucket1SnippetsTest/snippet-dependency-management-customizing-resolution-capability-substitution-rule_groovy_capability_substitution.sample)
    ```
    Could not get unknown property 'configurations' for task ':resolve' of type org.gradle.api.DefaultTask.
    ```

* `snippet-dependency-management-customizing-resolution-capability-substitution-rule_kotlin_capability`[(Build Scan)](https://ge.gradle.org/s/wkszl3ueqiln2/tests/:docs:docsTest/org.gradle.docs.samples.Bucket1SnippetsTest/snippet-dependency-management-customizing-resolution-capability-substitution-rule_kotlin_capability_substitution.sample)
    ```
    Task `:resolve` of type `org.gradle.api.DefaultTask`: cannot serialize Gradle script object references as these are not supported with the configuration cache.
    ```

### How to reproduce
To reproduce the failure in the Gradle documentation tests, run the following in the `gradle/gradle`repository:
```sh
./gradlew :docs:docsTest \
    -PrunBrokenConfigurationCacheDocsTests=true \
    -PenableConfigurationCacheForDocsTests=true \
    --tests "*snippet-dependency-management-customizing-resolution-capability-substitution-rule_groovy_capability_substitution.sample" \
    --tests "*snippet-dependency-management-customizing-resolution-capability-substitution-rule_kotlin_capability_substitution.sample"
```
In the test build outputs, which you can find in the test report, the configuration cache problems are listed along with the URL pointing to the detailed HTML problems report that can help investigating the violation. 
You can find more instructions on working with the sample projects in the [FAQ](https://github.com/gradle/cc-hackathon-2022/blob/main/faq.md).
### Docs including the snippets from this sample project
The following doc pages include the snippets from this sample project: 
- `subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/resolution_rules.adoc`;

### PR DESCRIPTION
**Pull request by cguerreros**
Adjust the snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule sample for configuration cache

Fixes #21803

### CODE REVIEW

File: subprojects/docs/src/snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule/groovy/build.gradle [hunk 1]
```diff
@@ -20,8 +20,8 @@ configurations.testCompileClasspath {
 // end::substitution_rule[]
 
 tasks.register('resolve') {
-    inputs.files(configurations.testCompileClasspath)
```
Comment (abstratt) [inline subprojects/docs/src/snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule/groovy/build.gradle:23]: We don't want to remove the declaration of this task's inputs.  Without the input being declared, the dependency between the `resolve` and the Java compilation tasks won't be identified by Gradle.

Other than that, your changes look good.

**Review by abstratt — CHANGES_REQUESTED** [review]

File: subprojects/docs/src/snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule/kotlin/build.gradle.kts [hunk 2]
```diff
@@ -20,8 +20,9 @@ configurations.testCompileClasspath {
 
 
 tasks.register("resolve") {
-    inputs.files(configurations.testCompileClasspath)
```
Comment (abstratt) [inline subprojects/docs/src/snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule/kotlin/build.gradle.kts:23]: same here

File: subprojects/docs/src/snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule/groovy/build.gradle (hunk 1 above)
Comment (cguerreros) [reply subprojects/docs/src/snippets/dependencyManagement/customizingResolution-capabilitySubstitutionRule/groovy/build.gradle:23]: @abstratt Sorry about this, I put back the line as it was before.

**Comment by cguerreros** [comment]
@abstratt requested changes done.

**Review by abstratt — APPROVED** [review]
Thanks!

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Comment by abstratt** [comment]
🎉

**Pull request merged**